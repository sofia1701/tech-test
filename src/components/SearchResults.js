import React from "react";
import "../styles/SearchResults.css";
import PropTypes from "prop-types";

const SearchResults = ({ results }) => {
  return (
    <div className="search-results">
      {results.map((img) => (
        <img src={img} alt="img" key={img} />
      ))}
    </div>
  );
};

SearchResults.propTypes = {
  results: PropTypes.array,
};

export default SearchResults;
