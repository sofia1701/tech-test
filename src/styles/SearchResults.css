.search-results {
  text-align: center;
  padding: 20px 20px;
}

img {
  width: 300px;
  height: 300px;
}