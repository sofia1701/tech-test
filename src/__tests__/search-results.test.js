import React from "react";
import { render } from "@testing-library/react";
import SearchResults from "../components/SearchResults";

describe("SearchResults", () => {
  const mockResults = [
    { img: "image1" }
  ];

  it("renders correctly", () => {
    const { asFragment } = render(<SearchResults results={mockResults} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
