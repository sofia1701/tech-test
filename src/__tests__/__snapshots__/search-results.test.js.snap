// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults renders correctly 1`] = `
<DocumentFragment>
  <div
    class="search-results"
  >
    <img
      alt="img"
      src="[object Object]"
    />
  </div>
</DocumentFragment>
`;
